File: app/src/main/java/com/sdex/activityrunner/db/history/HistoryModel.java
package com.sdex.activityrunner.db.history;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

@Entity
public class HistoryModel {

  @PrimaryKey(autoGenerate = true)
  private int id;
  private long timestamp;
  private String packageName;
  private String className;
  private int action;
  private String data;
  private int mimeType;
  private String categories;
  private String flags;

  public HistoryModel() {
  }

  public void setId(int id) {
    this.id = id;
  }

  public int getId() {
    return id;
  }

  public long getTimestamp() {
    return timestamp;
  }

  public void setTimestamp(long timestamp) {
    this.timestamp = timestamp;
  }

  public String getPackageName() {
    return packageName;
  }

  public void setPackageName(String packageName) {
    this.packageName = packageName;
  }

  public String getClassName() {
    return className;
  }

  public void setClassName(String className) {
    this.className = className;
  }

  public int getAction() {
    return action;
  }

  public void setAction(int action) {
    this.action = action;
  }

  public String getData() {
    return data;
  }

  public void setData(String data) {
    this.data = data;
  }

  public int getMimeType() {
    return mimeType;
  }

  public void setMimeType(int mimeType) {
    this.mimeType = mimeType;
  }

  public String getCategories() {
    return categories;
  }

  public void setCategories(String categories) {
    this.categories = categories;
  }

  public String getFlags() {
    return flags;
  }

  public void setFlags(String flags) {
    this.flags = flags;
  }
}

File: app/src/main/java/com/sdex/activityrunner/db/history/HistoryModel.kt
package com.sdex.activityrunner.db.history

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.io.Serializable

@Entity
data class HistoryModel(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    val timestamp: Long,
    val name: String?,
    val packageName: String?,
    val className: String?,
    val action: String? = null,
    val data: String? = null,
    val mimeType: String? = null,
    val categories: String? = null,
    val flags: String? = null,
    val extras: String? = null,
) : Serializable {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as HistoryModel
        return id == other.id
    }

    override fun hashCode(): Int {
        return id
    }
}
